packages/ckeditor5-dev-env/tests/translations/upload.js

```diff
@@ -10,67 +10,82 @@
 const path = require( 'path' );
 const sinon = require( 'sinon' );
 const mockery = require( 'mockery' );
-const transifexService = require( '../../lib/translations/transifex-service' );
-const fs = require( 'fs' );
 
 describe( 'upload', () => {
-	let sandbox;
+	const sandbox = sinon.sandbox.create();
 	let upload;
+	let packageNames;
+	let stubs;
 
 	beforeEach( () => {
-		sandbox = sinon.sandbox.create();
+		mockery.enable( {
+			useCleanCache: true,
+			warnOnReplace: false,
+			warnOnUnregistered: false,
+		} );
+
+		stubs = {
+			logger: {
+				info: sandbox.stub(),
+				warning: sandbox.stub(),
+				error: err => {
+					throw err;
+				},
+			},
+
+			transifexService: {
+				getResources: sandbox.spy( () => Promise.resolve( [ {
+					slug: 'ckeditor5-core'
+				} ] ) ),
+				postResource: sandbox.spy( () => Promise.resolve( '[]' ) ),
+				putResourceContent: sandbox.spy( () => Promise.resolve( '{}' ) ),
+			},
+
+			fs: {
+				readdirSync: sandbox.spy( () => packageNames ),
+				createReadStream: sandbox.spy( path => `${ path } content` ),
```

Comment:
[`fs.createReadStream`](https://nodejs.org/dist/latest-v6.x/docs/api/fs.html#fs_fs_createreadstream_path_options) returns an instance of ReadStream. It is not a string.